### ISSUE TITLE
creation of a tensor from a numba.cuda array

### ISSUE THREAD

**Issue by AK-ayush** [pytorch/pytorch#23067]
## 🚀 Feature

Having `torch.from_numba(cuda_arr)` can be useful in many cases.  
## Motivation
With the easy to use APIs in numba, I would like to do the preprocessing on gpu and then pass resultant gpu arrays to pytorch model. But currently, we have to convert these gpu arrays to numpy and then use the `torch.from_numpy()` API. 

## Additional context
```Python
import numba 
import torch
cuda_arr = numba.cuda.device_array(shape=(5,4))
# here, we need to perform this unnecessary conversion from gpu->cpu
numpy_arr = cuda_arr.copy_to_host()
gpu_tensor = torch.from_numpy(numpy_arr)
```

**Comment by AK-ayush** [comment]
I have a figured a temporary workaround for the time being but having a feature similar to above would be highly useful.

https://gist.github.com/AK-ayush/202875ef816025653980bd1d6e5a0071

**Comment by a-akram-98** [comment]
I will use this method in the middle of training there is a for loop that I will use numba on it in the middle of the training so I will convert the torch cuda to numba cuda and then convert it to torch cuda again will the autograd still report the gradients!?

**Comment by machineko** [comment]
@a-akram-98 You can check https://pytorch.org/docs/stable/dlpack.html u can get easy gpu->gpu without data copying but im not sure if gradient got calculated (it should as its just custom cuda operations)

**Comment by brunomagacho** [comment]
@AK-ayush this has been solved? I'm actually looking for something like this right now
